### PR TITLE
Create LassoCV-brankaj.py

### PR DESCRIPTION
**Pull request by brankaj**
This file contains Python implementation of the Lasso model. This is my first time using pull requests so I am not sure am I doing this right. Any feedback would be great!

### CODE REVIEW

**Comment by dhimmel** [comment]
Can you add `algorithms/LassoCV-brankaj.ipynb` to this pull request?

**Comment by brankaj** [comment]
I have added the file, but when I open it on github I cannot see the comments that I have made. Before each modified section I have added a comment. Maybe things should be ok if users open it in Jupyter Notebook.

**Comment by dhimmel** [comment]
The comments do show up [on nbviewer](http://nbviewer.jupyter.org/github/brankaj/machine-learning/blob/f505adf651292db44f638787913d235ac52aa5e3/algorithms/scripts/LassoCV-brankaj.ipynb). Note that GitHub does not render everything in notebooks for security reasons.

To create these comments, it looks like you used a markdown cell but then wrapped it between `<FONT COLOR=\"FF00FF\">\n` and `</FONT>`? Is that the case?

**Comment by dhimmel** [comment]
> In TCGA-MLexample.ipynb two methods are used when presenting output results: decision_function and predict_proba. Lasso model does not have method predict_proba, hence it is removed from the code. Additionally, we obtain a warning that decision_function is going to be deprecated in future versions.

I think the issue is that LassoCV is performing regression rather than logistic regression. In other words, it's not designed for classification. Rather it's designed for predicting a continuous outcome variable. Nevertheless, this pull request is valuable and the method seems to work, just that it's not "conventional" _per se_.

**Comment by dhimmel** [comment]
Everything looks good to me. I'm going to merge unless @brankaj has any changes she would like to make first.

**Comment by brankaj** [comment]
It's great that comments show up on nbviewer. I was searching for an option to change color in markdown cells directly, but the only thing I could find was using HTML tags. The only issue with this approach is that the color attribute is not supported in HTML5 and I would need different code for this.

Anyways, I do not plan to make any more changes in the current file. Thanks for pointing out the continuity requirement for LASSO. There are better ways to incorporate sparsity if we want it. Actually, since in other thread #31 you mentioned that 3 algorithms including LASSO would be implemented using SGDClassifier, I can play with the main code and see what I get when I use l1 penalty in the parameter list of the SGDClassifier. I should get similar results and report this here. 

Thanks for your feedback!

**Comment by dhimmel** [comment]
Merged. Congrats on your first pull request for Project Cognoma. 

> I can play with the main code and see what I get when I use l1 penalty in the parameter list of the SGDClassifier. I should get similar results and report this here.

Would be interesting to see!

**Comment by brankaj** [comment]
Thank you. I plan to report new results by tomorrow evening.

**Comment by brankaj** [comment]
Sorry for the late report, I was doing some other simulations on my computer. Since this thread is closed I am not going to submit new files, just to report scores from the heatmap (obtained from the section Process Mutation Matrix in the main code). 

Lasso model can be obtained using Elastic net with l1_ratio=1. If the regularization term is fixed (i.e., penalty is l1) we can change loss functions. Options for loss function are listed in http://scikit-learn.org/stable/modules/sgd.html#mathematical-formulation. I did not observe significant difference in the scores when using different cost functions (around 84.7%). Only when I increased the number of features to 1000 I got 87.1%. This is still lower than the score obtained when using additional regularization term with 500 features.
